xarray/core/dataset.py

```diff
@@ -6683,20 +6684,23 @@ def pad(
                     end_values=end_values,
                     reflect_type=reflect_type,
                 )
-            elif name in var_pad_width.keys() and not isinstance(
-                pad_width[name], int
-            ):  # dimension coordinates
-                w0, w1 = pad_width[name]  # type: ignore
-                fill_value_ind = dtypes.get_fill_value(var.dtype)
-                if isinstance(w0, int):
-                    w0_ = IndexVariable(name, [fill_value_ind] * w0)
-                else:
-                    w0_ = IndexVariable(name, w0)
-                if isinstance(w1, int):
-                    w1_ = IndexVariable(name, [fill_value_ind] * w1)
-                else:
-                    w1_ = IndexVariable(name, w1)
-                variables[name] = var.concat([w0_, var, w1_], dim=name)
+            elif (
+                name in var_pad_width.keys()  # dimension coordinates
+                and isinstance(pad_width[name], Sequence)
+                and (
+                    isinstance(pad_width[name][0], Sequence)  # type: ignore
+                    or isinstance(pad_width[name][1], Sequence)  # type: ignore
+                )
+            ):
+                pad_start, pad_end = pad_width[name]  # type: ignore
+                if isinstance(pad_start, int) or isinstance(pad_end, int):
+                    # do not allow [Sequence, int] as pad_width
+                    raise TypeError(
+                        "({}, {}) is used for pad_width[{}]. Must be either (int, int) or (Sequence, Sequence).".format(
```

Comment:
Noticed that a mixture of sequence and int for pad_width, like
```python
da.pad(dim2=([0, 1, 2], 3))
```
makes implementation very complex (as we may need to support many options, such as pad_mode).
I just disallowed the mixed argument.

Reply:
:+1: